torch/csrc/distributed/c10d/Backend.cpp

```diff
@@ -15,4 +32,26 @@ void Backend::init() {
   C10_LOG_API_USAGE_ONCE(fmt::format("c10d.backend_{}", getBackendName()));
 }
 
+void Backend::emitCollectiveStart(const Work& work) {
+  details::EventInfo evt;
+  commonEventinit(evt, *this, work);
+
+  evt.event_kind = ::c10d::EventKind::CollectiveStart;
+  details::enqueue_c10d_event(std::move(evt));
+}
+
+void Backend::emitCollectiveEnd(const Work& work) {
+  details::EventInfo evt;
+  commonEventinit(evt, *this, work);
+
+  evt.event_kind = ::c10d::EventKind::CollectiveEnd;
+  // FIXME change getDuration to return Optional<float>
+  try {
+    evt.duration_ms = work.getDuration();
+  } catch (std::exception& e) {
+    C10D_INFO("Duraction not available {}", e.what());
```

Comment:
hmm. this is probably not ok-

our 'disabled by default' approach for this feature applies only to the inner call to `details::enqueue_c10d_event(std::move(evt));`

however, even if that's disabled, we're still gonna swallow any exception that's raised from work.getDuration().  My concern is we'll prevent this error from being surfaced to the existing watchdog.  OR will it be re-raised on any subsequent access to work obj?

Reply:
There's a big todo for this part, it's due to an API design flaw, let me fix it.